R/sample.R

```diff
@@ -156,16 +212,61 @@ combine_chains <- function(res) {
                 "runner")))
   }
 
-  ret <- list(pars = pars,
-              density = density,
-              details = details,
-              chain = chain)
-  class(ret) <- "mcstate_samples"
-  ret
+  samples <- list(pars = pars,
+                  density = density,
+                  details = details,
+                  chain = chain)
+  class(samples) <- "mcstate_samples"
+  samples
+}
+
+
+## This is absolutely terrible, but it will get there.
+append_chains <- function(prev, curr) {
+  n_chains <- length(prev$restart$rng_state)
+  i <- split(seq_along(prev$chain), prev$chain)
+  j <- split(seq_along(curr$chain) + length(prev$chain), curr$chain)
+  j <- lapply(j, function(x) x[-1])
+  k <- unlist(rbind(i, j))
```

Comment:
This is just to get the the `chain` chain indexes lined up nicely for the appended chains?